pkgs/development/tools/repository-managers/nexus/default.nix

```diff
@@ -2,11 +2,11 @@
 
 stdenv.mkDerivation rec {
   pname = "nexus";
-  version = "3.45.0-01";
+  version = "3.52.0-01";
 
   src = fetchurl {
-    url = "https://sonatype-download.global.ssl.fastly.net/nexus/3/nexus-${version}-unix.tar.gz";
-    hash = "sha256-ISTORslFPgFt0kEKK17fpBXi3yAMwgm6qDnk33V2wa0=";
+    url = "https://download.sonatype.com/nexus/3/nexus-${version}-unix.tar.gz";
+    sha256 = "19gk79mjfhalsnr7hd20z4pzxbfx0jwchna0ih4da1nn5sxncxzq";
```

Comment:
Any reason for the switch back to the `sha256` hash style? I think the SRI-hash is the way forward. Not a blocker anyway.

Reply:
Ah yeah you right, I missed that. Usually we want to use hash with a self describing hash like before. They have the same effect, hash is just the preferred thing in nixpkgs.

Reply:
I switched to `sha256` because I couldn't get the `hash` approach to work. My understanding is that SRI hashes are base 64 encoded hashes with a prefix, but when I tried

```
hash = "sha256-MTlnazc5bWpmaGFsc25yN2hkMjB6NHB6eGJmeDBqd2NobmEwaWg0ZGExbm41c3huY3h6cQ==";
```

I get

```
 nix-build -A nexus --show-trace
error: invalid SRI hash 'MTlnazc5bWpmaGFsc25yN2hkMjB6NHB6eGJmeDBqd2NobmEwaWg0ZGExbm41c3huY3h6cQ=='                 

       … while evaluating the attribute 'src' of the derivation 'nexus-3.52.0-01'

       at /home/amy/code/nix/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:303:7:

          302|     // (lib.optionalAttrs (attrs ? name || (attrs ? pname && attrs ? version)) {
          303|       name =
             |       ^
          304|         let
```

where the above `sha256` value works. Any ideas what I'm not doing correctly to get the SRI hash value? I also tried with just the `19gk79mjfhalsnr7hd20z4pzxbfx0jwchna0ih4da1nn5sxncxzq` value just in case and got a similar error.

Reply:
Btw, if you have the experimental feature `flakes` enabled, any of the nix tools will output the SRI format when doing TOFU (trust on first use)